### ISSUE TITLE
TBSync periodically creates `Local Backup of: X` calendars, removing old calendar settings

### ISSUE THREAD

**Issue by venthur** [jobisoft/TbSync#130]
## Your environment

TbSync version: 0.7.15
Thunderbird version: 60

[ ] Yes, I have installed the latest available (beta) version from 
https://github.com/jobisoft/TbSync/releases 
and my issue is not yet fixed, I can still reproduce it.

## Expected behavior

Leave my calendars alone

## Actual behavior

Every week or so my TBSync'd calendars have different settings (i.e. the color setting) and new calendars appear: `Local Backup of: $Calendar`

## Steps to reproduce

Setup a calendar with TBSync and wait for a few weeks.

To help resolving your issue, enable debug logging (TbSync Account Manager -> Help) and send me the debug.log via e-mail (use the title of your issue as subject of the email).

**Comment by jobisoft** [comment]
Something is wrong, TbSync looses contact to the original caledar on the server. It will download the "new" calendar abd backup any old local "leftover" calendar.

What server are you connecting to? Horde?

**Comment by venthur** [comment]
It is an official microsoft server: https://outlook.office365.com

**Comment by jobisoft** [comment]
status?

**Comment by BillBrowne** [comment]
This has happened to me occasionally in the past and happened to me today. I was on the last debug release but just updated to 0.7.16.3 so am not sure what release I was on when it happened.

**Comment by jobisoft** [comment]
For every reporter in this issue: I need the following information per incident:
 - what type of server is that account connecting to
 - the additional info in the name of the backuped calendar/addressbook (for example "forced resync")

**Comment by BillBrowne** [comment]
The server I'm connected to is outlook.office365.com
Sorry, I deleted the back ups so cannot remember their names

**Comment by pcause** [comment]
I've had this as well.  I am on latest released (7.16).  The name of my backup was 

Local backup of: O365 Contacts [deleted from server] (*)

I am on a laptop, travel frequently and often get slow/intermittent connections to o365.  it's imap support is slow/bad.

**Comment by jobisoft** [comment]
@pcause when that happens, did ALL your folders (address books and calendars and tasks) pop off like that, or just some?

**Comment by pcause** [comment]
just my o365 address book,

**Comment by jobisoft** [comment]
This is important, so let me rephrase that: Did all folders you subscribed to using the tbsync manager pop off at the same time, or just one and the others remained?

**Comment by pcause** [comment]
Ok, not 100% sure what you are asking.  I have only one o365 account.  My calendar seemed to still be there/fine and I don't use tasks in o365.  mail folders (imap) all seemed fine

**Comment by jobisoft** [comment]
![image](https://user-images.githubusercontent.com/5830621/47033788-ff55ec00-d175-11e8-9fc9-47b648ce0d38.png)
As you can see, I have two folders subscribed. I do not know how many you have subscribed. I am asking, if this issue happens to all your subscribed folders at once, or if only one gets popped off (with a backup)

**Comment by pcause** [comment]
I have Contacts, Calendar and Tasks subscribed.

**Comment by jobisoft** [comment]
Thanks for your feedback. Could you check in lightning, if your calendar and task folder have also been backuped with [deleted on server]?

**Comment by jobisoft** [comment]
This issue was moved to jobisoft/EAS-4-TbSync#18

**Comment by tchuddi** [comment]
Hi, this actually happend just now to me on 2 PCs and with 2 exchange account with all callendars.
Thunderbird 60.6.1; TBsync 1.7
both accounts are exchange. one is office365 the second is outlook.com
Local backup of: calendar_name (email@outlook.com) [forced folder resync]
right now i have like 50 of these "backups" :\

**Comment by PresGas** [comment]
I can confirm this is happening to me as well (what @tchuddi said above).

**Comment by jobisoft** [comment]
Can you provide a log, where I can see, why there is a forced folder resync? 

Enable debug logging under help in the tbsync account manager, restart Thunderbird, sync a couple of times and as soon as there is a new backup, go back to the help tab and create a bug report. Thanks!

**Comment by tchuddi** [comment]
well, i enabled debug, but i am using TBsync for very long time and this is actually first time that happend to me. If it ever happend again I will post it
I found this thread when i looked how to erase all of those "backups".

**Comment by PresGas** [comment]
Bug report(s) sent.

**Comment by jobisoft** [comment]
Ah sorry, that log does not contain the information I need. To make sure that users do not send me personal or confidential data with those logs, the default setting is to include only error messages. You need to set a userdatalevel in the advanced settings of Thunderbird, to increase the amount of data written into the log.

However, I just learned, that the steps of those levels are to big. The first level which includes the information I need is 2, but currently that sends ALL your personal data as well. Which is not good.

I just released a new beta, which sends only the basic sync data but not any of your contact or calendar data, if the userdatalevel is set to 2.

So I need to ask you to:

a) [update](https://github.com/jobisoft/TbSync/wiki/Get-the-latest-TbSync-version#geting-updates-via-the-beta-release-channel) or [install ](https://github.com/jobisoft/TbSync/wiki/Get-the-latest-TbSync-version#beta-release-channel) the beta of TbSync

b) open the advanced Settings of Thunderbird and set

`extensions.tbsync.log.userdatalevel = 2`

c) restart Thunderbird

d) send a new log if the issue happens again (you can open the log attached to the email to see what is send)

Sorry for the inconvenience and thanks for your support.

**Comment by joolsr** [comment]
I've had the same thing happening for me a couple of times now . the mysterious 'local backup of X' calendars breaking the calendar sync. I've followed your notes above and will create a bug report as It would be great to get this fixed ;-)

**Comment by jobisoft** [comment]
Did you TB just update to TB 60.7?

**Comment by joolsr** [comment]
Using Nextcloud Server : 15.0.7
Thunderbird : 60.6.1
: Zorin 16,04

I followed your steps carefully as above https://github.com/jobisoft/TbSync/issues/130#issuecomment-489763644 but when I inspected the error log I can also see carddav private content ie names and addresses so can't send the log as is at the moment. I'm very happy to help debug this problem so we can get this bug fixed ;-)

I think the jist of the problem is that I am using two different caldav/card-dav accounts which are shared on the Nextcloud server which was the only way I could get two accounts to show in Lightning

**Comment by joolsr** [comment]
hi @jobisoft No still using 60.6

**Comment by jobisoft** [comment]
Let me rephrase: Is it possible, that TB updating itself could be related to the issue?

**Comment by jobisoft** [comment]
Ah ok, thanks.

**Comment by jobisoft** [comment]
You only get this error for calendars? Not address books?

**Comment by joolsr** [comment]
i can clear the issue temporarily by unsubscribing the calendar and contacts for the users account that is at fault (there are two, but one is fine), then resubscribing selecting the required calendar and Contacts again

**Comment by joolsr** [comment]
i havent checked address books yet

**Comment by jobisoft** [comment]
If it is only calendars, please do the steps described here and check if that makes theses errors stop:
https://github.com/jobisoft/DAV-4-TbSync/wiki/Adding-multiple-accounts-from-the-same-server

I have a patch pending at TB to get that fixed, so that workaround is no longer needed.

**Comment by joolsr** [comment]
in fact at the moment when i switch on the calendar it lights up ok for about half a second then it goes off and goes grey again. The NC server calendats are working fine with Evolution

**Comment by joolsr** [comment]
Hi again. I've just done what you suggest with the adding multiple accounts from same server, so fingers crossed

**Comment by joolsr** [comment]
I've done the steps as in multiple accounts on same server and its working now. I have a feeling i may have done this previously anyway but will need to check. Thanks for your help

**Comment by jobisoft** [comment]
This is the patch for the issue you see. I hope to get it into TB soon. I hope it will and before/with TB68:
https://bugzilla.mozilla.org/show_bug.cgi?id=1544596

**Comment by csmdbr** [comment]
I have the same problem. My configuration : 

- Thunderbird 60.7.0
- Exchange server (OVH provider)
- TBSync Beta 1.7.6
- Provider for Exchange ActiveSync Beta 0.12.7

Configured on 6 different computers (Windows & Mac)

Each time TBSync synchronized with the server, local back of Calendars AND Address Book appear. The reminders also popup after each sync.

Please help this problem is very serious and urgent. Thanks you very much !

I followed your instruction to create a log. I attach it.

[debug.log](https://github.com/jobisoft/TbSync/files/3273247/debug.log)

**Comment by csmdbr** [comment]
Update : it seems there are several problems around this issue : 

- Local backups are created almost at each synchronization
- After a time, calendars and contacts selected from an account are just turned off (going to settings of TBSync, the calendar and contact list of the account has been unticked)
- Local modifications of calendars are not pushed on the server when the automatic synchronization takes place ! Appointment just disappear, which is a very serious issue...

I don't know if they are all related, but there is definitely a cascade failures there...

**Comment by jobisoft** [comment]
That means, folders change their UID between syncs. This is very strange and should not happen. Maybe OVH knows more? TbSync provides a very detailed error log (if `extensions.tbsync.log.userdatalevel` is set to 9) to see what is happening. 

I cannot investigate this within the next 4 weeks, as all my time goes into release of update for Thunderbird 68.

But I will look into this.

**Comment by jobisoft** [comment]
Please recheck this with TB68.

**Comment by MatthiasCarnein** [comment]
I can confirm that TB68 fixed this issue for me. Thank you!
